src/main/java/com/google/api/codegen/metacode/InitCodeNode.java

```diff
@@ -387,12 +387,31 @@ private void resolveSampleParamConfigs(
       fieldPath = parentFieldPath + "." + key;
     }
     this.sampleParamConfig = sampleParamConfigMap.get(fieldPath);
-    if (sampleParamConfig != null && sampleParamConfig.readFromFile()) {
-      setLineType(InitCodeLineType.ReadFileInitLine);
+
+    if (sampleParamConfig != null) {
+
+      // If read_file is set to true in config, change the line type to ReadFileInitLine.
+      if (sampleParamConfig.readFromFile()) {
+        setLineType(InitCodeLineType.ReadFileInitLine);
+      }
+
+      // If sample_argument_name is specified in config, set identifier to this name if the name has
+      // not be used yet and error out otherwise.
```

Comment:
Why is it an error? Shouldn't we just rename it like we do other variables?

Reply:
Hmm, I would lean towards erroring rather than automatically changing a variable name the user has explicitly specified

Reply:
That makes sense to me too.